sqlmesh/core/test/definition.py

```diff
@@ -82,10 +82,17 @@ def setUp(self) -> None:
             rows = values["rows"]
             if not columns_to_types and rows:
                 for i, v in rows[0].items():
-                    # convert ruamel into python
-                    v = v.real if hasattr(v, "real") else v
-                    v_type = annotate_types(exp.convert(v)).type or type(v).__name__
-                    columns_to_types[i] = exp.maybe_parse(v_type, into=exp.DataType)
+                    if isinstance(v, dict):
+                        # Since a YAML dict can be interpreted in many ways (e.g. STRUCT, MAP),
+                        # we parse it to figure out what the correct representation is per dialect
+                        v_ast: exp.Expression = exp.maybe_parse(str(v), dialect=self.dialect)
```

Comment:
i don't understand how this works. what is v? how can you parse it as sql?

Reply:
Afaik the possible values in the yaml format are strings, numerics, booleans, null, and lists/dicts that can be nested and contain such values. These should all be parseable as sql.

Reply:
We've discussed about this and decided to follow a different approach, so I'm temporarily making this a draft.